code/obj/item/Bamboo_spike_trap.dm

```diff
@@ -0,0 +1,207 @@
+			return
+		if (src.armed)
+			set_icon_state("Bamboo_Spike_Trap-Packet")
+			logTheThing(LOG_COMBAT, User, "disarmed a spike trap at [src.loc]")
+			src.armed = FALSE
+			src.anchored = FALSE
+		return
+
+	proc/break_down(mob/User)
+		//breaks down into multiple bamboo parts
+		if (!src)
+			return
+		var/obj/item/material_piece/organic/bamboo/A = new /obj/item/material_piece/organic/bamboo(get_turf(src))
+		if (src.break_down_amount > 1)
+			A.change_stack_amount(src.break_down_amount - 1)
+		A.add_fingerprint(User)
+		qdel(src)
+
+
+
+	hitby(atom/movable/AM, datum/thrown_thing/thr)
+		..()
+		if (src.armed && ishuman(AM) && AM.throwing)
+			var/mob/living/carbon/human/victim = AM
+			//crashes into the trap when being thrown/slipped at it
+			victim.visible_message("<span class='alert'><B>[victim] crashes into the spike trap!</B></span>",\
+			"<span class='alert'><B>You crash into the spike trap!</B></span>")
+			crash_into(victim)
+			qdel(src) //if crashed into, destroys the trap
+			return
+
+	attack_self(mob/User as mob)
+		if (!src.armed)
+			var/trap_occupied = 0
+			for(var/obj/item/B in get_turf(src))
+				if (istype(B, /obj/item/bamboo_spike_trap))
+					var/obj/item/bamboo_spike_trap/BM = B
+					if (BM.armed)
+						trap_occupied = 1
+			if (!trap_occupied)
+				User.show_text("You start to arm the trap...", "blue")
+				var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(User, src, src.arming_time,
+				/obj/item/bamboo_spike_trap/proc/arm,\list(User), src.icon, src.icon_state, "[User] finishes arming [src]")
+				actions.start(action_bar, User)
+				return
+			else
+				boutput(User, "<span class='alert'>A trap is already placed here!</span>")
+		..()
+
+
+	Crossed(atom/movable/AM as mob|obj)
+		..()
+		if (src.armed && ishuman(AM))
+			var/mob/living/carbon/human/victim = AM
+			//crawling or just walking between the sticks is a viable counter
+			//getting thrown at the trap has a different effect we want to check seperately
+			if(victim.lying || victim.throwing || !src.checkRun(victim))
+				return
+			//If any checks failed, well, you step into the trap
+			victim.visible_message("<span class='alert'><B>[victim] steps into the spike trap!</B></span>",\
+			"<span class='alert'><B>You step into the spike trap!</B></span>")
+			step_on(victim)
+
+	proc/crash_into(mob/living/carbon/human/victim as mob)
+		if (!src || !victim || !src.armed)
+			return
+		logTheThing(LOG_COMBAT, victim, "crashed into [src] at [log_loc(src)].")
+		victim.changeStatus("stunned", src.crashed_stun)
+		victim.force_laydown_standup()
+		src.trap_damage(victim, src.crashed_force)
+		playsound(victim.loc, 'sound/impact_sounds/Generic_Hit_Heavy_1.ogg', 80, 1)
+		victim.UpdateDamageIcon()
+
+
+	proc/step_on(mob/living/carbon/human/victim as mob)
+		if (!src || !victim || !src.armed)
+			return
+		logTheThing(LOG_COMBAT, victim, "stepped into [src] at [log_loc(src)].")
+		victim.changeStatus("weakened", src.armed_weakened)
```

Comment:
stunned and weakened do exactly the same thing here with the same duration, curious why you're using two different effects?

Reply:
If they do, i should probably make some changes to the duration. I thought with weakened here you are able to move while forced on the ground (honestly, completely forgot to test that)

Reply:
Yeah, they are working completely the same. I make them both the weakened condition and reduce the duration when stepped into.